From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct vsock_sock
Patch-mainline: never, kabi
References: kabi

In netfilter-stable-18_08_17, commit
455f05ecd2b219e9a216050796d30c830d9bc393 (vsock: split dwork to avoid
reinitializations) splitted one delayed work into two. It made the kABI
checker complaining.

Given the structure is allocated by __vsock_create, move one of the
members to the end and pretend nothing really happened.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/net/af_vsock.h |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

--- a/include/net/af_vsock.h
+++ b/include/net/af_vsock.h
@@ -62,8 +62,11 @@ struct vsock_sock {
 	struct list_head pending_links;
 	struct list_head accept_queue;
 	bool rejected;
-	struct delayed_work connect_work;
+#ifdef __GENKSYMS__
+	struct delayed_work dwork;
+#else
 	struct delayed_work pending_work;
+#endif
 	struct delayed_work close_work;
 	bool close_work_scheduled;
 	u32 peer_shutdown;
@@ -72,6 +75,9 @@ struct vsock_sock {
 
 	/* Private to transport. */
 	void *trans;
+#ifndef __GENKSYMS__
+	struct delayed_work connect_work;
+#endif
 };
 
 s64 vsock_stream_has_data(struct vsock_sock *vsk);
